src/core/utils/channel_manager.hpp

```diff
@@ -78,6 +85,19 @@ class ChannelManager : public InstanceLocator, private NonCopyable
      */
     explicit ChannelManager(Instance &aInstance);
 
+    /**
+     * Requests a channel change.
+     *
+     * The channel changes depending on the device role. On Full Thread Devices, a network channel change is requested
+     * (`RequestNetworkChannelChange`). On Minimal Thread Devices with an active csl link, the csl channel changes
+     * (`ChangeCslChannel`).
+     *
+     * @param[in] aChannel             The new channel.
+     *
+     */
+    void RequestChannelChange(uint8_t aChannel);
```

Comment:
Should we change this (or remove this)?

Reply:
Made `private` for use with auto select timer.